domain.tf

```
@@ -0,0 +1,10 @@
+data "ns_domain" "this" {
+  stack_id = data.ns_workspace.this.stack_id
+  block_id = data.ns_workspace.this.block_id
+}
+
+resource "google_dns_managed_zone" "this" {
+  name     = data.ns_workspace.this.block_name
```

```suggestion
  name     = data.ns_workspace.this.block_name
```
I assume `name` is the unique identifier for managed zones.
Let's use `block_ref` to ensure uniqueness.

---

Yes, that name should be unique within the project. Taken care.